### DISCUSSION TITLE
feat: generative user interface

### DISCUSSION THREAD

**Discussion by dannyl1u** [open-webui/open-webui#5593] [Ideas]
https://github.com/miurla/morphic
https://morphic.sh/

Morphic is a "An AI-powered answer engine with a generative UI." This would be a cool addition to Open-webui.

**Describe the solution you'd like**
A toggle switch to enable/disable this feature.

**Additional context**
<img width="1624" alt="image" src="https://github.com/open-webui/open-webui/assets/45186464/79d34c0f-409f-4067-aade-a56704047e9a">

**Comment by silentoplayz** [comment]
So something like [Perplexity.AI](https://www.perplexity.ai/), but as a toggle-able feature for Open WebUI? Or if we're strictly talking something open-sourced like you have mentioned, there's also this I've stumbled across: [ClarityAI](https://github.com/mckaywrigley/clarity-ai).

Honorable mention: [Omniplex](https://omniplex.vercel.app/) https://www.reddit.com/r/LocalLLaMA/comments/1bg3ug0/working_on_opensource_perplexity_ai/
